docs/timezone.md

```diff
@@ -27,6 +27,12 @@ The object must have `getOffsetNanosecondsFor()`, `getPossibleInstantsFor()`, an
 It must not have a `timeZone` property, so that it can be distinguished in `Temporal.TimeZone.from()` from other Temporal objects that have a time zone.
 It is possible to pass such an object into any Temporal API that would normally take a built-in `Temporal.TimeZone`.
 
+Any implementation of the protocol with those three methods will return the correct output from any Temporal property or method.
```

Comment:
I'm confused about the content above what's changed in the PR. Why are required methods different between the subclass implementation and the custom-object implementation?  

Also, the TS types also list `getDateTimeFor` as required.  Which is correct?

Reply:
The subclass allows you to skip implementing both `id` and `toString()` by virtue of letting you set the [[Identifier]] internal slot, by calling `super(id)` in the constructor. But note that might go away depending on the outcome of #780 

You're right though, `getNextTransition()` and `getPreviousTransition()` should be documented to be optional in the subclass case as well. I'll fix this up tomorrow.